### PR TITLE
[Typecript] Convert current

### PR DESCRIPTION
**Pull request by Hinton**

### CODE REVIEW

File: src/popup/app/current/current.component.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        });
+    }
+
+    private loadVault() {
+        chrome.tabs.query({ active: true, currentWindow: true }, (tabs: any) => {
+            if (tabs.length > 0) {
+                this.url = tabs[0].url;
+            } else {
+                this.$timeout(() => {
+                    this.loaded = true;
```
Comment (kspearrin) [inline src/popup/app/current/current.component.ts]: same

File: src/popup/app/current/current.component.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        }
+
+        this.autofillService
+            .doAutoFill({
+                cipher,
+                pageDetails: this.pageDetails,
+                fromBackground: false,
+            })
+            .then((totpCode: string) => {
+                this.$analytics.eventTrack('Autofilled');
```
Comment (kspearrin) [inline src/popup/app/current/current.component.ts]: this is correct in this scope?

Comment (Hinton) [reply src/popup/app/current/current.component.ts]: Yea, uses arrow syntax which binds this. I also fixed the scope issue with callback.

File: src/popup/app/current/current.component.ts (hunk 1 above)
Comment (Hinton) [reply src/popup/app/current/current.component.ts]: Yes.

**Comment by kspearrin** [comment]
Didnt know that about arrow functions. Cool

File: src/popup/app/current/current.component.ts [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    otherCiphers: any = [];
+    loginCiphers: any = [];
+    url: any;
+    domain: any;
+    canAutofill: boolean = false;
+    searchText: string = null;
+    inSidebar: boolean = false;
+    disableSearch: boolean = false;
+
+    constructor($scope: any, private cipherService: any, private utilsService: UtilsService, private toastr: any,
```
Comment (kspearrin) [inline src/popup/app/current/current.component.ts]: Scope can be private?

File: src/popup/app/current/current.component.ts [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+        this.i18n = i18nService;
+        this.inSidebar = utilsService.inSidebar($window);
+        this.disableSearch = utilsService.isEdge();
+
+        document.getElementById('search').focus();
+
+        $scope.$on('syncCompleted', (event: any, successfully: any) => {
+            if (this.loaded) {
+                setTimeout(this.loadVault.bind(this), 500);
```
Comment (kspearrin) [inline src/popup/app/current/current.component.ts]: use `$timeout`?

File: src/popup/app/current/current.component.ts [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+                private $analytics: any, private i18nService: any, private constantsService: any,
+                private $filter: any) {
+
+        this.i18n = i18nService;
+        this.inSidebar = utilsService.inSidebar($window);
+        this.disableSearch = utilsService.isEdge();
+
+        document.getElementById('search').focus();
+
+        $scope.$on('syncCompleted', (event: any, successfully: any) => {
```
Comment (kspearrin) [inline src/popup/app/current/current.component.ts]: successfully is a boolean i believe

File: src/popup/app/current/current.component.ts (hunk 3 above)
Comment (Hinton) [reply src/popup/app/current/current.component.ts]: It's only used in the constructor, so no need to assign it as a member.

**Pull request merged**